src/routes/docs/tutorials/nextjs/step-4/+page.markdoc

```diff
@@ -0,0 +1,219 @@
+---
+layout: tutorial
+title: Implement authtentication
+description: Add Appwrite authentication to your Next.js app
+step: 4
+---
+
+# Signup page {% #signup-page %}      
+
+In your project under the app directory, create a new file `src/app/(pages)/signup/page.js`. Next.js will automatically create a route for this page `/signup`. Let's create the signup form.
+
+```js
+"use client";
+
+import { useState } from "react";
+import Link from "next/link";
+import toast from "react-hot-toast";
+import { useRouter } from "next/navigation";
+import { ID } from "appwrite";
+import { useCookie } from "next-cookie";
+
+import { account } from "@/config/appwrite";
+
+const SignUp = (props) => {
+  const [name, setName] = useState("");
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
```

Comment:
We don't need to use state variables for forms in this case, you can use the [formData on the submit function](https://react.dev/reference/react-dom/components/input#reading-the-input-values-when-submitting-a-form)